published/009-2016-March-28.md

```diff
@@ -0,0 +1,219 @@
+
+Many people jumped to IPFS as a possible solution to this problem. With a permanent filesystem, unpublishing wouldn't be possible. Here's one post titled [How to use IPFS to fix npm](http://amreldib.com/blog/HowToUseIpfsToFixNpm); here's [an issue](https://github.com/ipmjs/ipmjs/issues/9) on a new GitHub organization, ipmjs, trying to find consensus on how to fix npm using a permanent storage system; here's an npm module, [cowpen](https://www.npmjs.com/package/cowpen) that publishes modules directly to IPFS; here's [another decentralized package manager](https://github.com/mhhf/spore) that sprung up using IPFS and Ethereum.
+
+The IPFS community has been thinking about immutable package managers for a long time. @diasdavid has a project called `registry-mirror`, which allows you to run an npm registry locally that is backed by packages retrieved from IPFS instead of NPM directly. He's written about a presentation he gave for it, [here](http://blog.daviddias.me/2015/12/08/stellar-module-management); the source code is [here](https://github.com/diasdavid/registry-mirror).
+
+On a similar note, [gx](//github.com/whyrusleeping/gx), a package manager for Go, was also mentioned in a lot of the discussions about npm and package managers, especially on [Hacker News](https://news.ycombinator.com/). In the past two weeks, the project has recieved over 1000 stars - before, there were around 50.
+
+The discussion about how to best use IPFS as a package manager is ongoing. Jump on GitHub if you have something to say; we're listening in the [FAQ](https://github.com/ipfs/faq/issues) and in [the notes repo](https://github.com/ipfs/notes/issues).
+
+### DNS outage
+
+We're using DigitalOcean for providing ipfs.io DNS. On Tuesday DigitalOcean DNS was hit by a [severe outage](https://www.digitalocean.com/company/blog/update-on-the-march-24-2016-dns-outage/) lasting hours, which took the public gateway at ipfs.io down. We switched to [DNSimple](https://dnsimple.com) in an ad-hoc fashion and brought ipfs.io back while DigitalOcean was still down, but this incident obviously hit us on the wrong foot a bit. We'll be working to never get taken down this way again. Here's a few things we'll do:
+
+- codified DNS zones, and tools to upload them to DNS providers
+- one or two backup DNS providers
+- monitoring and failover procedures
+
+We'll post a more detailed postmorten on our blog in the next few days.
+
+### Captain.log
+
+... TODO @diasdavid
```

Comment:
### Aye, you might want to check the new js-ipfs Captain.log entry, mate!

Following [js-ipfs roadmap](https://github.com/ipfs/js-ipfs/blob/master/ROADMAP.md#milestone-1---js-ipfs-on-the-browser), we’re close™ to have a workable js-ipfs version that can be used in the browser and in Node.js. This will mark a very important milestone on the IPFS project and enable a whole set of new distributed web applications to be possible. If you want to be part of this effort, check out our [Captain.log](https://github.com/ipfs/js-ipfs/issues/30#issuecomment-205884015) entry to get a full update and a list of tasks you can contribute to.